### ISSUE TITLE
Unnecessary parentheses - false positive

### ISSUE THREAD

**Issue by bazzik** [kalessil/phpinspectionsea#641]
Not sure if this is already covered by #579 ...

- PhpStorm 2017.2.4
- Php Inspections (EA extended) 2.3.12.1

```
$a = ($b = function($c) { return $c ;})(1);
$z = $b(2);
echo $a, "\n";
echo $z, "\n";
```

Original code sample taken from https://youtrack.jetbrains.com/issue/WI-38778

**Comment by kalessil** [comment]
Fixed
